### PR TITLE
Fix GradientClipByGlobalNorm dtype bug

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

Bug fixes
### PR changes

APIs
### Describe

Fix GradientClipByGlobalNorm dtype bug.

```
import paddle
import numpy as np
paddle.disable_static()

paddle.set_default_dtype('float64')

linear = paddle.nn.Linear(10, 10)
print(linear.weight.dtype)
print(linear.bias.dtype)

out = linear(paddle.to_tensor(np.random.rand(10, 10)))
out.backward()

optimizer = paddle.optimizer.Adam(0.1, parameters=linear.parameters(), grad_clip=paddle.nn.GradientClipByGlobalNorm(1.0))
optimizer.step()
```

image

### CODE REVIEW

File: python/paddle/fluid/clip.py [hunk 1]
```diff
@@ -590,7 +591,9 @@ def _dygraph_clip(self, params_grads):
         global_norm_var = layers.reduce_sum(global_norm_var)
         global_norm_var = layers.sqrt(global_norm_var)
         max_global_norm = layers.fill_constant(
-            shape=[1], dtype='float32', value=self.clip_norm)
```
Comment (zhiqiu) [inline python/paddle/fluid/clip.py:593]: Better infer dtype by `global_norm_var`?

File: python/paddle/fluid/clip.py [hunk 2]
```diff
@@ -635,7 +638,9 @@ def _static_clip(self, params_grads):
                 global_norm_var = layers.sums(sum_square_list)
                 global_norm_var = layers.sqrt(x=global_norm_var)
                 max_global_norm = layers.fill_constant(
-                    shape=[1], dtype="float32", value=self.clip_norm)
```
Comment (zhiqiu) [inline python/paddle/fluid/clip.py:638]: Same above.

Comment (zhwesky2010) [reply python/paddle/fluid/clip.py:638]: Ok

File: python/paddle/fluid/clip.py (hunk 1 above)
Comment (zhwesky2010) [reply python/paddle/fluid/clip.py:593]: Done

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**